src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,196 @@
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        MyNode<T> node = findNodeByIndex(index);
+        unlink(node);
+        return node.item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        MyNode<T> node;
+        if ((node = findNodeByValue(object)) == null) {
+            return false;
+        }
+        unlink(node);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder output = new StringBuilder();
+        MyNode<T> current = first;
+        while (current != null) {
+            if (current != first) {
+                output.append(", ");
+            }
+            output.append(current.item);
+            current = current.next;
+        }
+        return String.format("[%s] (%d)", output, size());
+    }
+
+    private boolean isElementIndex(int index) {
+        return index >= 0 && index < size;
+    }
+
+    private String outOfBoundsMsg(int index) {
+        return String.format("Index: %d, Size: %d", index, size);
+    }
+
+    private void checkElementIndex(int index) {
+        if (!isElementIndex(index)) {
+            throw new IndexOutOfBoundsException(outOfBoundsMsg(index));
+        }
+    }
+
+    private MyNode<T> findNodeByValue(T value) {
+        MyNode<T> current = first;
+        while (current != null) {
+            if (value == current.item || (value != null && value.equals(current.item))) {
+                return current;
+            }
+            current = current.next;
+        }
+        return null;
+    }
+
+    private MyNode<T> findNodeByIndex(int index) {
+        return (index < size / 2) ? findNodeByIndexFromHead(index) : findNodeByIndexFromTail(index);
+    }
+
+    private MyNode<T> findNodeByIndexFromHead(int index) {
+        MyNode<T> current = first;
+        int currentIndex = 0;
+        while (current != null) {
+            if (currentIndex == index) {
+                return current;
+            }
+            currentIndex++;
+            current = current.next;
+        }
+        throw new IndexOutOfBoundsException(outOfBoundsMsg(index));
+    }
+
+    private MyNode<T> findNodeByIndexFromTail(int index) {
+        MyNode<T> current = last;
+        int currentIndex = size - 1;
+        while (current != null) {
+            if (currentIndex == index) {
+                return current;
+            }
+            currentIndex--;
+            current = current.prev;
+        }
+        throw new IndexOutOfBoundsException(outOfBoundsMsg(index));
+    }
+
+    private void unlink(MyNode<T> node) {
+        if (node == null) {
+            return;
+        }
+        size--;
```

Comment:
```suggestion
        size--;
```
Let’s  move it to the end of the method